pkgs/by-name/wi/winbox4/package.nix

```diff
@@ -20,21 +21,35 @@ stdenvNoCC.mkDerivation (finalAttrs: {
   pname = "winbox";
   version = "4.0beta16";
 
-  src = fetchurl {
-    name = "WinBox_Linux-${finalAttrs.version}.zip";
-    url = "https://download.mikrotik.com/routeros/winbox/${finalAttrs.version}/WinBox_Linux.zip";
-    hash = "sha256-RZpsKew3BaId6+tcwUV6fniUpCH4wIP9ab6P5oE7OAk=";
-  };
+  src =
+    if stdenvNoCC.isDarwin then
+      fetchurl {
+        name = "WinBox-${finalAttrs.version}.dmg";
+        url = "https://download.mikrotik.com/routeros/winbox/${finalAttrs.version}/WinBox.dmg";
+        hash = "sha256-ZxvuEVx9BmFukPMEPKeqXQNW38ExSpnRcSuHdw6j+CI=";
+      }
+    else
+      fetchurl {
+        name = "WinBox_Linux-${finalAttrs.version}.zip";
+        url = "https://download.mikrotik.com/routeros/winbox/${finalAttrs.version}/WinBox_Linux.zip";
+        hash = "sha256-RZpsKew3BaId6+tcwUV6fniUpCH4wIP9ab6P5oE7OAk=";
+      };
 
   sourceRoot = ".";
 
-  nativeBuildInputs = [
-    autoPatchelfHook
-    copyDesktopItems
-    # makeBinaryWrapper does not support --run
-    makeWrapper
-    unzip
-  ];
+  nativeBuildInputs =
+    if stdenvNoCC.isDarwin then
```

Comment:
Instead of using the classical binary statement `if...then...else`, can you instead use the `lib.optionals <condition> [ <stuff> ]` statement?